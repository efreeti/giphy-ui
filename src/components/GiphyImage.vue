<template>
  <span>
    <img v-if="visible" class="md-image" :src="imageInfo.url" :width="imageInfo.width" :height="imageInfo.height"/>
    <div v-else role="placeholder" :style="{ width: `${imageInfo.width}px`, height: `${imageInfo.height}px` }"
         v-observe-visibility="{ callback: handleVisibility, once: true }">
    </div>
  </span>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { GiphyImageInfo } from '@/models/GiphyImageInfo';

@Component
export default class GiphyImage extends Vue {
  @Prop() imageInfo!: GiphyImageInfo;

  private visible: boolean = false;

  handleVisibility(isVisible: boolean) {
    this.visible = this.visible || isVisible;
  }
}
</script>
