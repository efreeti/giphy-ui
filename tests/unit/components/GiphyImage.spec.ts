import Vue from 'vue';
import { mount } from '@vue/test-utils';
import { intersectionObserver } from '@shopify/jest-dom-mocks';

import '../config';

import GiphyImage from '@/components/GiphyImage.vue';


describe('GiphyImage', () => {
  beforeEach(() => {
    intersectionObserver.mock();
  });

  afterEach(() => {
    intersectionObserver.restore();
  });

  it('should render uninitialized correctly', () => {
    expect(mount(GiphyImage, {
      propsData: {
        imageInfo: {
          url: 'some-image.gif',
          width: 100,
          height: 100,
        },
      },
    }))
        .toMatchSnapshot();
  });

  it('should render initialized correctly', async () => {
    const wrapper = mount(GiphyImage, {
      propsData: {
        imageInfo: {
          url: 'some-image.gif',
          width: 100,
          height: 100,
        },
      },
    });

    await Vue.nextTick();

		intersectionObserver.simulate({
			target: wrapper.find('[role=placeholder]').element,
			isIntersecting: true,
		});

		expect(wrapper).toMatchSnapshot();
  });
});
