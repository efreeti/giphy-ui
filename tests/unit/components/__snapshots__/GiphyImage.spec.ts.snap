// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyImage should render initialized correctly 1`] = `<span><img src="some-image.gif" width="100" height="100" class="md-image"></span>`;

exports[`GiphyImage should render uninitialized correctly 1`] = `<span><div role="placeholder" style="width: 100px; height: 100px;"></div></span>`;
